source/Mission.cpp

```diff
@@ -968,7 +968,9 @@ Mission Mission::Instantiate(const PlayerInfo &player) const
 		destinations.erase(bestIt);
 	}
 	DistanceMap distance(source);
-	jumps += distance.Days(result.destination->GetSystem());
+	if (result.destination) {
```

Comment:
mission.cpp:886-892 verifies result.destination is not nullptr, so this change is not needed.
Your change also does not meet the style guide.